src/test/scala/is/hail/methods/TableSuite.scala

```diff
@@ -59,13 +60,41 @@ class TableSuite extends SparkSuite {
     val importedData = sc.hadoopConfiguration.readLines(inputFile)(_.map(_.value).toIndexedSeq)
     val exportedData = sc.hadoopConfiguration.readLines(outputFile)(_.map(_.value).toIndexedSeq)
 
-    intercept[HailException] {
-      hc.importTable(inputFile).keyBy(List("Sample", "Status", "BadKeyName"))
+    intercept[AssertionError] {
+      hc.importTable(inputFile).keyBy("Sample", "Status", "BadKeyName")
     }
 
     assert(importedData == exportedData)
   }
 
+  @Test def testWriteReadOrdered() {
+    val outputFile = tmpDir.createTempFile("ktRdWrtOrd")
+    sampleKT1.write(outputFile)
+    val read = Table.read(hc, outputFile)
+
+    assert(read.rvd.isInstanceOf[OrderedRVD])
+    assert(read.same(sampleKT1))
+  }
+
+  @Test def testWriteReadUnordered() {
+    val outputFile = tmpDir.createTempFile("ktRdWrtUnord")
+    sampleKT1.unkey().write(outputFile)
+    val read = Table.read(hc, outputFile)
+
+    assert(read.rvd.isInstanceOf[UnpartitionedRVD])
+    assert(read.same(sampleKT1.unkey()))
+  }
+
+  @Test def testKeyBy() {
+    val kt = sampleKT1
+    val count = kt.count()
+    val kt2 = kt.keyBy(Array("Sample", "field1"), Array("Sample"))
+    assert(kt2.count() == count)
+    assert(kt2.keyBy(Array("Sample"), Array("Sample")).count() == count)
+    assert(kt.keyBy("field1").count() == count)
+    assert(kt.unkey().keyBy(Array("Sample")).count() == count)
```

Comment:
They aren't the same, because they have different keys. Really I wanted to exercise different paths of changing from one key to another (key by a prefix of the old key, which requires no sorting, key by an extension of the old key, which requires only local sorting, etc.), making sure all the code paths in `keyBy` are being run, mainly to be sure it executes and all assertions pass. The count is just to force evaluation. I'm happy to discuss more, as I had trouble figuring out how to test this, but I still don't see a better option.